src/main/java/randoop/main/GenInputsAbstract.java

```diff
@@ -1147,6 +1173,145 @@ private static boolean shouldOmitClass(String classname) {
+                            && accessibility.isAccessible(Class.forName(classname));
+                      } catch (ClassNotFoundException e) {
+                        throw new RandoopClassNameError(
+                            classname,
+                            String.format(
+                                "Cannot load class found in directory %s",
+                                directory.getAbsolutePath()));
+                      }
+                    })
+                .collect(Collectors.toSet());
+        return classnames;
+      } catch (IOException e) {
+        throw new RandoopUsageError(
+            String.format(
+                "Caught an exception while reading directory %s", directory.getAbsolutePath()),
+            e);
+      }
+    }
+    return Collections.emptySet();
+  }
+
+  /**
+   * Returns classes with the given package in the given jar file.
+   *
+   * @param jarFile a jar file containing classes
+   * @param packageName a package name
+   * @param accessibility the accessibility predicate
+   * @return classes in package {@code packageName} in the given jar file
+   */
+  public static Set<@ClassGetName String> searchInJar(
+      File jarFile, String packageName, AccessibilityPredicate accessibility) {
+    Set<@ClassGetName String> classnames = new HashSet<>();
+    @ClassGetName String classname = "randoop.main.GenInputsAbstract";
+    try (ZipInputStream zip = new ZipInputStream(new FileInputStream(jarFile.toString()))) {
+      for (ZipEntry entry = zip.getNextEntry(); entry != null; entry = zip.getNextEntry()) {
+        String entryName = entry.getName();
+        if (!entry.isDirectory() && entryName.endsWith(".class")) {
+          @SuppressWarnings("signature")
+          @InternalForm String ifClassName = entryName.substring(0, entryName.length() - ".class".length());
+
+          classname = Signatures.internalFormToClassGetName(ifClassName);
+          if (classname.startsWith(packageName)
+              && !classname.substring(packageName.length() + 1).contains(".")
+              && accessibility.isAccessible(Class.forName(classname))) {
+            classnames.add(classname);
+          }
+        }
+      }
+    } catch (FileNotFoundException e) {
+      throw new RandoopUsageError(
+          String.format(
+              "Cannot find .jar file specified in classpath: %s", jarFile.getAbsolutePath()));
+    } catch (IOException e) {
+      throw new RandoopUsageError(
+          String.format("Cannot read .jar file: %s", jarFile.getAbsolutePath()));
+    } catch (ClassNotFoundException e) {
+      throw new RandoopClassNameError(
+          classname, String.format("Cannot load class found in %s", jarFile.getAbsolutePath()));
+    }
+    return classnames;
+  }
+
+  /**
+   * Returns list of jars and directories, present in classpath.
+   *
+   * @return List of jars and directories, present in classpath
+   */
+  public static List<File> getClassLocationsForCurrentClasspath() {
```

Comment:
Creating a list is not necessary.  This method can return an array, eliminating the need for the loop and the `urls` variable.  (Then, you might find the method so small that you can inline it.)

Reply:
Good idea! I inlined it and moved creation of File object to loop which called that method, so new list is not created now.

Reply:
Thanks for addressing my feedback.  I appreciate it.  As a general rule, if you accepted a suggestion, you don't need to respond.  (This advice is from 
https://homes.cs.washington.edu/~mernst/advice/github-pull-request.html#respond-to-reviews .)